### PR TITLE
feat: scaffold solution template entity

### PR DESCRIPTION
**Pull request by juliannemarik**
### Description:
Scaffolds the Hub Solution Template entity

note: we made the decision to simply call this entity a "Hub Template"/`type = "template"` rather than "Hub Solution Template"/`type = "solutionTemplate"`

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)
1. [x] used semantic commit messages
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)
1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/templates/HubTemplate.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  deleteTemplate,
+  editorToTemplate,
+  updateTemplate,
+} from "./edit";
+import { fetchTemplate } from "./fetch";
+import { IEntityEditorContext } from "../core/types";
+import { enrichEntity } from "../core/enrichEntity";
+import { cloneObject } from "../util";
+
+/** Hub Template Class */
```
Comment (dbouwman) [inline packages/common/src/templates/HubTemplate.ts:19]: Maybe add more explanation here that this is used for all Solution items despite the Hub team not "owning" that item type. Main goal is to facilitate editing the meta info (the "item"), managing sharing etc

File: packages/common/src/templates/HubTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+   * @param context
+   * @param save
+   */
+  static async create(
+    partialTemplate: Partial<IHubTemplate>,
+    context: IArcGISContext,
+    save: boolean = false
+  ): Promise<HubTemplate> {
+    const pojo = this.applyDefaults(partialTemplate, context);
+    const instance = HubTemplate.fromJson(pojo, context);
```
Comment (dbouwman) [inline packages/common/src/templates/HubTemplate.ts:64]: Should this throw if we don't want to allow creating them from Hub?

File: packages/common/src/templates/HubTemplate.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  /** Save the HubTemplate to the backing store */
+  async save(): Promise<void> {
+    if (this.isDestroyed) {
+      throw new Error("HubTemplate is already destroyed.");
+    }
+
+    // 1. create or update
+    this.entity = this.entity.id
+      ? await updateTemplate(this.entity, this.context.userRequestOptions)
+      : await createTemplate(this.entity, this.context.userRequestOptions);
```
Comment (dbouwman) [inline packages/common/src/templates/HubTemplate.ts:207]: Similar to above - should we create new Solution items? If we allow it, someone will do it :)

File: packages/common/src/templates/_internal/TemplateSchema.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { IConfigurationSchema } from "../../core";
+import { HubItemEntitySchema } from "../../core/schemas/shared/HubItemEntitySchema";
+
+export type TemplateEditorType = (typeof TemplateEditorTypes)[number];
+
+export const TemplateEditorTypes = ["hub:template:edit"] as const;
+
+export const TemplateSchema: IConfigurationSchema = {
+  ...HubItemEntitySchema,
```
Comment (dbouwman) [inline packages/common/src/templates/_internal/TemplateSchema.ts:9]: Really like how these are getting quite concise :)

File: packages/common/src/templates/utils.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * is supposed to become when activated. This util
+ * extracts the expected item type from the
+ * hubSolutionType typeKeyword on the solution item
+ *
+ * We use this in various UIs to show the expected
+ * type rather than "Solution"
+ *
+ * @param template
+ */
+export const getActivatedTemplateType = (template: IItem): string => {
```
Comment (dbouwman) [inline packages/common/src/templates/utils.ts:16]: Wonder if we should call this `getDeployedTemplateType` ... or something using the term "deployed" vs "Activated", as it's more aligned with Solutions terminology (I believe... at least they add a Deployed keyword...)

File: packages/common/src/templates/HubTemplate.ts (hunk 2 above)
Comment (juliannemarik) [reply packages/common/src/templates/HubTemplate.ts:70]: I was considering this as well. I think that's a good call. I've updated the underlying `createTemplate` function to throw an error and added comments to the class methods that call that function letting users know we don't currently support creation

**Review by jordantsanz — APPROVED** [review]
Looks good, nice work! 🚀

**Pull request merged**